core/lib/resources/presence/sentry.js

```diff
@@ -1,159 +1,270 @@
+
+  if (typeof(options) === 'function') { 
+    callback = options; 
+  } else {
+    this._applyOptions(options); 
   }
-  if (!this.subscribed) {
-    this.subscribed = true;
-    Persistence.pubsub().subscribe(Sentry.channel);
+
+  if (this._refreshTimer) { return; }
+
+  logging.info('#presence - #sentry - starting', this.name);
+
+  if (options.expiration) {
+    keepAliveOptions.expiration = options.expiration;
   }
+
+  upMessage = self._keepAlive(keepAliveOptions);
+
+  this.setMaxListeners(0);
+  this._loadAndCleanUpSentries(function() {
+    self.emit('up', this.name, upMessage);
+    Persistence.publish(redisSentriesKey, upMessage);
+    self._startListening();
+    if (callback) { callback(); }
+  });
+
+  this._refreshTimer = setTimeout(this._refresh.bind(this), Math.floor(this._refreshInterval/2));
+  this._checkSentriesTimer = setTimeout(this._checkSentries.bind(this), Math.floor(this._checkInterval/2));
 };
 
+Sentry.prototype.stop = function(callback) {
+  logging.info('#presence- #sentry stopping', this.name);
+  
+  this._stopTimer('_checkSentriesTimer');
+  this._stopTimer('_refreshTimer');
+  this.sentries = {};
+  this._stopListening();
+
+  if (callback) { callback(); }
+};
 
-Sentry.prototype._save = function(message) {
-  if (message && message.name) {
-    logging.debug('#presence - sentry.save', message.name,
-                  (message.expiration - Date.now()) + '/' + Sentry.expiry);
-    this.sentries[message.name] = message;
+Sentry.prototype.sentryNames = function() {
+  return Object.keys(this.sentries);
+};
+
+Sentry.prototype.isDown = function(name) {
+  var lastMessage = this.sentries[name],
+      isSentryDown = messageIsExpired(lastMessage);
+
+  if (isSentryDown) {
+    var expiration = messageExpiration(lastMessage); 
+    var text = expiration ? expiration + '/' + this._defaultExpiryOffset : 'not-present';
+    logging.debug('#presence - #sentry isDown', name, isSentryDown, text);
   }
+
+  return isSentryDown;
 };
 
-Sentry.prototype._sentryDown = function(name) {
-  if (!this.isValid(name)) {
+Sentry.prototype._setName = function(name) {
+  this.name = name || this._generateName();
+  return this.name;
+};
 
-    // Cleanup
-    Persistence.deleteHash(Sentry.channel, name);
+Sentry.prototype._generateName = function() {
+  var shasum = require('crypto').createHash('sha1'),
+      newName;
 
-    var details = this.sentries[name];
-    logging.info('#presence - #sentry down:', name,
-                    details.host, details.port);
-    delete this.sentries[name];
-    this.emit('down', name, details);
+  shasum.update(require('os').hostname() + ' ' + Math.random() + ' ' + Date.now());
+  newName = shasum.digest('hex').slice(0,15);
+  
+  return newName;
+};
+
+Sentry.prototype._applyOptions = function(options) {
+  if (options) {
+    this.host = options.host;
+    this.port = options.port;
+   
+    this._defaultExpiryOffset = options.defaultExpiryOffset || defaultOptions.DEFAULT_EXPIRY_OFFSET;
```

Comment:
Same goes for default everywhere for this. Because you are not using default anywhere else and that's fine. (defaultOptions explains it well that everything inside it is default)